### PR TITLE
improve specification

### PR DESCRIPTION
**Pull request by ankithans**
- Add architecture-related details in the readme.
- update registry.json

links to #1

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,41 +1,61 @@
 # Charmil
 
+> TODO Logo
```
Comment (wtrocki) [inline README.md:3]:
```suggestion
```

File: README.md [hunk 2]
```diff
@@ -1,41 +1,61 @@
...
+
 Framework for building command line plugins on top of Golang Cobra Library that allows developers to install and use other (even non golang CLIs) as extensions.
 Charmil will let you control your cobra based CLIs and extend them without embeeding any plugin code into your CLI.
 
 > NOTE: Under construction
```
Comment (wtrocki) [inline README.md:8]:
```suggestion
> NOTE:  Project is still under construction
```

File: README.md [hunk 3]
```diff
@@ -1,41 +1,61 @@
...
 
+## Working
```
Comment (wtrocki) [inline README.md:10]: Working?

File: README.md [hunk 4]
```diff
@@ -1,41 +1,61 @@
...
-`$ yourCLi extensions remove <name>@<version>`
+Include Charmil in your CLI by doing `$ go get -u github.com/aerogear/charmil`
 
+#### Commands that will be added to your CLI by Charmil
 
-## Getting started
+1. List all the available extensions
+   > `$ yourCLi extensions list`
+2. List all the installed extensions
+   > `$ yourCLi extensions installed`
```
Comment (wtrocki) [inline README.md:31]: Those will not be the commands. Please review commands that kew provides:
https://github.com/kubernetes-sigs/krew

Idealy we could keep the same pattern.

**Comment by wtrocki** [comment]
Every architecture diagram needs source (what tool you using for diagrams)
Let's create docs folder with images and sources so we can update them etc.

File: README.md [hunk 5]
```diff
@@ -1,41 +1,61 @@
...
+
 ## Architecture
 
-![architecture](mockups/architecture.png)
+1. Downloader pkg - It will download the binaries of plugins. Binaries may be zip / tar.gz. We need to extract them.
+2. Installer pkg - It will download and install a plugin. It will also support upgrade and remove plugins.
+3. Index pkg - Get All indexes, add, update or remove a index.
+4. Search Package - Search for packages via name, description locally and globally.
+
+![architecture](mockups/architecture3.png)
```
Comment (wtrocki) [inline README.md:61]: That will  be typically in documentation for the project

File: README.md (hunk 3 above)
Comment (ankithans) [reply README.md:10]: like how charmil CLI manager will work

explained via diagram

**Comment by ankithans** [comment]
> Every architecture diagram needs source (what tool you using for diagrams)

I am using draw.io 
> Let's create docs folder with images and sources so we can update them etc.

okay!

**Review by wtrocki — APPROVED** [review]
Let's get that in, however there is much more work on this - possibly we need docs for architecture and other things so readme is clear about what you do

**Pull request merged**